File: Selenium/Project/Activity9.java
package Activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.Assert;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;

import java.util.List;

public class Activity9 {
    private WebDriver driver;
    private Actions act;
    private WebDriverWait wait;

    @BeforeClass
    public void setUp(){
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE,"/dev/null");
        WebDriverManager.firefoxdriver().setup();
        driver = new FirefoxDriver();
        act = new Actions(driver);
        driver.get("http://alchemy.hguy.co/orangehrm");

    }
    @Test
    public void Login(){
        WebElement username = driver.findElement(By.id("txtUsername"));
        WebElement password = driver.findElement(By.id("txtPassword"));
        WebElement button = driver.findElement(By.id("btnLogin"));
        username.sendKeys("orange");
        password.sendKeys("orangepassword123");
        button.click();
        driver.manage().window().maximize();
        WebElement dashboard = driver.findElement(By.id("menu_dashboard_index"));
        Assert.assertTrue(dashboard.isDisplayed(),"Homepage is not displayed");

    }
    @Test
    public void retrieveEmergencyContact(){
        //wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//a[contains(@href,'viewMyDetails')]")));
        driver.findElement(By.xpath("//a[contains(@href,'viewMyDetails')]")).click();
        List<WebElement> sideNavigationMenu = driver.findElements(By.xpath("//ul[@id='sidenav']/li"));
        for (WebElement menuitems : sideNavigationMenu){
            String MenuList = menuitems.getText();
            if(MenuList.equals("Emergency Contacts")){
                menuitems.click();
                break;
            }

        }
        List<WebElement> tableRows = driver.findElements(By.xpath("//table[@id='emgcontact_list']/tbody/tr"));
        for(WebElement rowData : tableRows ){
             String contactList = rowData.getText();
            System.out.println("Contents in Emergency contact list is " + contactList);
        }

    }
    @AfterClass
    public void tearDown(){
        driver.quit();

    }
}

File: Python/Activities/Activity9.py
from selenium import webdriver
from selenium.webdriver.firefox.service import Service as FirefoxService
from  webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.common.by import By
from selenium.webdriver.common.action_chains import ActionChains
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.support.wait import WebDriverWait
from selenium.webdriver.support import expected_conditions


service = FirefoxService(GeckoDriverManager().install())

with webdriver.Firefox(service=service) as driver:
    driver.get("https://training-support.net/selenium/ajax")
    print("title of the page is :", driver.title)
    driver.find_element(By.XPATH,"//button[text()='Change Content']").click()
    wait = WebDriverWait(driver,5)
    wait.until(expected_conditions.text_to_be_present_in_element((By.TAG_NAME,"h1"),"HELLO!"))
    helloText = driver.find_element(By.TAG_NAME,"h1")
    print("text on clicking context is ", helloText.text)
    wait.until(expected_conditions.text_to_be_present_in_element((By.TAG_NAME, "h3"), "I'm late!"))
    late_text = driver.find_element(By.TAG_NAME, "h3").text
    # Print the text
    print(late_text)
    driver.quit()
